### PR TITLE
executor,mysql: fix #4540, Navicat for MySQL compability of show create table

### LINKED ISSUE #4540: navicat 11 `show create table` truncate
we have a table name t1:

```
CREATE TABLE `t1` (
  `a` char(36) NOT NULL,
  `b` timestamp NOT NULL DEFAULT CURRENT_TIMESTAMP,
  `c` timestamp NOT NULL DEFAULT CURRENT_TIMESTAMP,
  `d` varchar(50) DEFAULT '',
  `e` char(36) NOT NULL DEFAULT '',
  `f` char(36) NOT NULL DEFAULT '',
  `g` char(1) NOT NULL DEFAULT 'N',
  `h` varchar(100) NOT NULL,
  `i` timestamp NOT NULL DEFAULT CURRENT_TIMESTAMP,
  `j` varchar(10) DEFAULT '',
  `k` varchar(10) DEFAULT '',
  `l` varchar(20) DEFAULT '',
  `m` varchar(20) DEFAULT '',
  `n` varchar(30) DEFAULT '',
  `o` varchar(100) DEFAULT '',
  `p` varchar(50) DEFAULT '',
  `q` varchar(50) DEFAULT '',
  `r` varchar(100) DEFAULT '',
  `s` varchar(20) DEFAULT '',
  `t` varchar(50) DEFAULT '',
  `u` varchar(100) DEFAULT '',
  `v` varchar(50) DEFAULT '',
  `w` varchar(300) NOT NULL,
  `x` varchar(250) DEFAULT '',
  PRIMARY KEY (`a`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8 COLLATE=utf8_bin
```
we always set flen of `show create table` to 256, in most client, it will not use the flen. But in older navicat version (older than navicat 12), it will cause `show create table` truncated:

```
/*
Navicat MySQL Data Transfer

Source Server         : tidb
Source Server Version : 50701
Source Host           : 192.168.59.105:4000
Source Database       : test

Target Server Type    : MYSQL
Target Server Version : 50701
File Encoding         : 65001

Date: 2017-09-15 17:50:55
*/

SET FOREIGN_KEY_CHECKS=0;

-- ----------------------------
-- Table structure for t1
-- ----------------------------
DROP TABLE IF EXISTS `t1`;
CREATE TABLE `t1` (
  `a` char(36) NOT NULL,
  `b` timestamp NOT NULL DEFAULT CURRENT_;

-- ----------------------------
-- Records of t1
```

### PR DESCRIPTION
**Pull request by winkyao**
fix #4540 

@coocood @XuHuaiyu PTAL

### CODE REVIEW

**Comment by coocood** [comment]
If we multiply the charset max character bytes here

https://github.com/pingcap/tidb/blob/master/server/column.go#L55

It will also apply to general table columns.

File: mysql/charset.go [hunk 1]
```diff
@@ -558,6 +558,11 @@ const (
 	BinaryCollationID    = 63
 	UTF8DefaultCollation = "utf8_bin"
 	DefaultCollationName = UTF8DefaultCollation
+
+	// MaxBytesOfCharacter, is the max bytes length of a character,
+	// refer to RFC3629, In UTF-8, characters from the U+0000..U+10FFFF range
```
Comment (XuHuaiyu) [inline mysql/charset.go]: s/ In/ in

File: executor/show.go [hunk 2]
```diff
@@ -541,6 +541,25 @@ func (e *ShowExec) fetchShowCreateTable() error {
...
+	// the truncation, in MySQL, it will multiply bytes length by a multiple based on character set.
+	// For examples:
+	// * latin, the multiple is 1
+	// * gb2312, the multiple is 2
+	// * Utf-8, the multiple is 3
+	// * utf8mb4, the multiple is 4
+	// So the large enough multiple is 4 in here.
+	schema := e.Schema()
+	// Table | Create Table
+	if len(schema.Columns) == 2 {
```
Comment (XuHuaiyu) [inline executor/show.go]: why we need to check this ?

File: mysql/const.go [hunk 3]
```diff
@@ -59,11 +59,26 @@ const (
 )
 
 // Identifier length limitations.
+// See https://dev.mysql.com/doc/refman/5.7/en/identifiers.html.
```
Comment (XuHuaiyu) [inline mysql/const.go]: we do not need the `.` at the end for a comment like `See url`

**Comment by breeswish** [comment]
Currently we do not multiply flen with MaxBytesOfCharacter, otherwise a lot of function will got incorrect flen.

**Comment by winkyao** [comment]
@breeswish Please see the new update.

**Comment by winkyao** [comment]
@coocood @XuHuaiyu PTAL

File: server/server_test.go [hunk 4]
```diff
@@ -715,6 +715,52 @@ func runTestStmtCount(t *C) {
 	})
 }
 
+func (ts *TidbTestSuite) TestShowCreateTableFlen(c *C) {
+	// issue #4540
+	ctx, err := ts.tidbdrv.OpenCtx(uint64(0), 0, uint8(tmysql.CollationNames["utf8_general_ci"]), "test", nil)
```
Comment (coocood) [inline server/server_test.go]: We only support `utf8_bin` for now.
And you can use `DefaultCollationID`

**Comment by coocood** [comment]
LGTM

**Pull request merged**